plugins/optimization-detective/detect.js

```diff
@@ -161,6 +161,10 @@ async function getAlreadySubmittedSessionStorageKey(
 	currentUrl,
 	urlMetricGroupStatus
 ) {
+	if ( ! window.crypto || ! window.crypto.subtle ) {
+		throw new Error( 'Web Crypto API is unavailable' );
+	}
+
```

Comment:
```suggestion
```

Reply:
Alternatively:

```suggestion
	if ( ! window.crypto || ! window.crypto.subtle ) {
		warn( 'Unable to create session storage key: Web Crypto API is not available. This API is only available in secure contexts (HTTPS). Detection cannot proceed. If you are testing locally, ensure you use HTTPS or run on localhost.' );
		return 'odSubmitted';
	}

```

And then the rest of this function can be wrapped in a `try`/`catch` where in the `catch` block it returns `'odSubmitted'` after emitting a warning.

Reply:
But no, see https://github.com/WordPress/performance/pull/1911#discussion_r1989754696. It can just return `null` then.